### PR TITLE
topology: sof-adl-rt5682: add new topology

### PR DESCRIPTION
**Pull request by brentlu**
Add support for boards without speaker amplifier. Just simply remove
all speaker related components and configuration.

Signed-off-by: Brent Lu <brent.lu@intel.com>

### CODE REVIEW

**Comment by brentlu** [comment]
Linux PR: https://github.com/thesofproject/linux/pull/3371

File: tools/topology/topology1/CMakeLists.txt [hunk 1]
```diff
@@ -145,6 +145,7 @@ set(TPLGS
 	"sof-tgl-max98357a-rt5682\;sof-adl-max98390-rt5682\;-DCODEC=MAX98390\;-DFMT=s32le\;-DPLATFORM=adl\;-DAMP_SSP=1\;-DBT_OFFLOAD"
 	"sof-tgl-max98357a-rt5682\;sof-adl-max98390-rt5682-rtnr\;-DCODEC=MAX98390\;-DFMT=s32le\;-DPLATFORM=adl\;-DAMP_SSP=1\;-DBT_OFFLOAD\;-DCHANNELS=2\;-DRTNR"
 	"sof-tgl-max98357a-rt5682\;sof-adl-max98390-ssp2-rt5682-ssp0\;-DCODEC=MAX98390\;-DFMT=s32le\;-DPLATFORM=adl\;-DAMP_SSP=2"
+	"sof-tgl-max98357a-rt5682\;sof-adl-rt5682\;-DNO_AMP\;-DPLATFORM=adl"
```
Comment (lgirdwood) [inline tools/topology/topology1/CMakeLists.txt:148]: Is rt5682 the AMP ? if so, does it make more sense to have a new topology file ?

Comment (brentlu) [reply tools/topology/topology1/CMakeLists.txt:148]: nope. there is no speaker amplifier so I just add a compile option to remove all speaker related pipeline, dai, sst config...

**Review by lgirdwood — APPROVED** [review]

**Comment by sathyap-chrome** [comment]
@brentlu please raise PR for merge to adl-004-drop-stable as we are moving to adl004. thanks

**Pull request merged**